pkg/etcd/etcd.go

```diff
@@ -1011,5 +1011,13 @@ func (e *ETCD) GetMembersClientURLs(ctx context.Context) ([]string, error) {
 
 // RemoveSelf will remove the member if it exists in the cluster
 func (e *ETCD) RemoveSelf(ctx context.Context) error {
-	return e.removePeer(ctx, e.name, e.address, true)
+	if err := e.removePeer(ctx, e.name, e.address, true); err != nil {
+		return err
+	}
+
+	// backup the data dir to avoid issues when re-enabling etcd
+	oldDataDir := etcdDBDir(e.config) + "-old-" + strconv.Itoa(int(time.Now().Unix()))
+
+	// move the data directory to a temp path
+	return os.Rename(etcdDBDir(e.config), oldDataDir)
```

Comment:
Is there anything we should do in the event we're unable to remove the directory?

Reply:
In the case of not enabling to rename the dir it will return an error, I dont think we should proceed  if there is an error because it may consist an inconsistency